File: python/credit_check.py
def credit_check(inputPara):
    
    str_list = list(inputPara)
    final_sum = 0
    used_number = len(str_list)
    
    
    for prop in str_list:
            if int(prop) * used_number > 9:
                multipled_number = int(prop) * used_number
                multipled_number = str(multipled_number)
                split_num = []
                split_num.append(multipled_number[0])
                split_num.append(multipled_number[1])
                final_sum += int(split_num[0]) + int(split_num[1])
                

    if final_sum % 10 == 0:
        return "The number is valid!"
    else:
        return "The number is invalid!"

   


# print(credit_check("32325939"))

File: js/creditCheck.js
exports.creditCheck = function(str) {

    let strArr = str.split("");
    let finalSum = 0;
// This is the function for checking for numbers that are greater than 9, and adding to the sum
    for (let i = strArr.length - 2; i >= 0; i-- ){
        if (strArr[i] * 2 > 9){
            let multipliedNumber = strArr[i] * 2;
            let splitArr = multipliedNumber.toString().split("");
            finalSum += (parseInt(splitArr[0]) + parseInt(splitArr[1]))
            console.log(splitArr)
            console.log(finalSum);
         }
        //else{
        //     finalSum += parseInt(strArr[i] * 2)
        //     console.log(finalSum)
        // }
    }
    
    // //This deals with odd numbers that don't get duplicated
    for (i = 1; i < 14; i += 2){
        console.log(strArr[i])
        finalSum += parseInt(strArr[i]);
        console.log(strArr[i], finalSum)
     }
    finalSum += parseInt(strArr[15]);
    console.log(finalSum);
    
    
            if (finalSum % 10 == 0){
                return "The number is valid!";
            }else{
                return "The number is invalid!";
            }
    
    
    
    
    // for (let i = strArr.length - 3; i >= 0; i -= 2){
    //     if (strArr[i] * strArr[strArr.length - 1] > 9){
    //         let multipliedNumber = strArr[i] * strArr[strArr.length - 1];
    //         let splitArr = multipliedNumber.toString().split("");
    //         finalSum += (splitArr[0] + splitArr[1]);
    //     }
    // }
    //     if (finalSum % 10 == 0){
    //         return "The number is valid!";
    //     }else{
    //         return "The number is invalid!";
    //     }


}
console.log(exports.creditCheck("5541808923795240"));
